SecretSanta/src/SecretSanta.Api/Controllers/BaseApiController.cs

```diff
@@ -35,13 +38,18 @@ public async Task<IActionResult> Get(int id)
         }
 
         [HttpPut("{id}")]
-        public async Task<TEntity?> Put(int id, [FromBody] TEntity value)
+        public async Task<IActionResult> Put(int id, [FromBody] TInputDto value)
         {
-            return await Service.UpdateAsync(id, value);
+            TDto tDto = await Service.UpdateAsync(id, value);
```

Comment:
In general when naming you don't include the "t" from the generic parameter. 
```suggestion
            TDto dto = await Service.UpdateAsync(id, value);
```